lib/__tests__/pipeline.test.ts

```diff
@@ -92,60 +95,61 @@ test('can add arbitrary shellables with different artifacts', () => {
   }).action;
 
   pipeline.addPublish(new Pub(stack, 'Pub'), { inputArtifact: action.actionProperties.outputs![0] });
+  const template = Template.fromStack(stack);
 
-  cdk_expect(stack).to(haveResourceLike('AWS::CodePipeline::Pipeline', {
-    Stages: [
+  template.hasResourceProperties('AWS::CodePipeline::Pipeline', {
+    Stages: Match.arrayWith([
       {
-        Name: 'Source',
         Actions: [
-          {
-            ActionTypeId: { Category: 'Source', Owner: 'AWS', Provider: 'CodeCommit' },
+          Match.objectLike({
+            ActionTypeId: { Category: 'Source', Owner: 'AWS', Provider: 'CodeCommit', Version: '1' },
             Name: 'Pull',
             OutputArtifacts: [
               {
                 Name: 'Source',
               },
             ],
-          },
+          }),
         ],
+        Name: 'Source',
       },
       {
-        Name: 'Build',
         Actions: [
-          {
+          Match.objectLike({
             Name: 'Build',
-            ActionTypeId: { Category: 'Build', Owner: 'AWS', Provider: 'CodeBuild' },
+            ActionTypeId: { Category: 'Build', Owner: 'AWS', Provider: 'CodeBuild', Version: '1' },
             InputArtifacts: [{ Name: 'Source' }],
             OutputArtifacts: [{ Name: 'Artifact_Build_Build' }],
             RunOrder: 1,
-          },
+          }),
         ],
+        Name: 'Build',
       },
       {
-        Name: 'Test',
         Actions: [
-          {
-            ActionTypeId: { Category: 'Build', Owner: 'AWS', Provider: 'CodeBuild' },
+          Match.objectLike({
+            ActionTypeId: { Category: 'Build', Owner: 'AWS', Provider: 'CodeBuild', Version: '1' },
             InputArtifacts: [{ Name: 'Artifact_Build_Build' }],
             Name: 'ActionSecondStep',
-            OutputArtifacts: [{ Name: 'Artifact_TestStackPipelineSecondStep91726C11' }],
+            OutputArtifacts: [{ Name: 'Artifact_c81eddcbe9657bd312a728fb13df77bc09f9a519b4' }],
             RunOrder: 1,
-          },
+          }),
         ],
+        Name: 'Test',
       },
       {
-        Name: 'Publish',
         Actions: [
-          {
-            ActionTypeId: { Category: 'Build', Owner: 'AWS', Provider: 'CodeBuild' },
-            InputArtifacts: [{ Name: 'Artifact_TestStackPipelineSecondStep91726C11' }],
+          Match.objectLike({
+            ActionTypeId: { Category: 'Build', Owner: 'AWS', Provider: 'CodeBuild', Version: '1' },
+            InputArtifacts: [{ Name: 'Artifact_c81eddcbe9657bd312a728fb13df77bc09f9a519b4' }],
```

Comment:
The artifact name change is what we get when we go from `node.uniqueId` => `node.addr`